### PR TITLE
Update to SCTK 0.18.0

### LINKED ISSUE #34: Significant power usage in applications using this crate
The implementation of this crate relies on a sleep loop to operate. However, that means it will also cause the CPU to wake up frequently even if there are no clipboard traffic of any kind in the system.

### LINKED ISSUE #44: [FR] please update to to wayland* 0.30
Hi, the current version of the rust wayland stack is 0.30. Please consider updating this crate so it's on par with the rest. I tried updating it but it results in a lot of unresolved imports.

### LINKED ISSUE #46: alacritty warning queue destroyed while proxies still attached
Alacritty prints warning on exit:

warning: queue 0x7fc404000cc0 destroyed while proxies still attached:
  zwp_primary_selection_offer_v1@4278190081 still attached
  wl_data_offer@4278190080 still attached
  zwp_primary_selection_device_v1@29 still attached
  zwp_primary_selection_device_manager_v1@25 still attached
  wl_data_device@28 still attached
  wl_data_device_manager@27 still attached
  wl_seat@26 still attached
  wl_registry@24 still attached

Thread 3 "smithay-clipboa" hit Breakpoint 1, 0x00007ffff7fb6e20 in wl_event_queue_destroy () from /usr/lib64/libwayland-client.so
(gdb) bt
#0  0x00007ffff7fb6e20 in wl_event_queue_destroy () from /usr/lib64/libwayland-client.so
#1  0x0000555555c842f6 in wayland_client::imp::event_queue::{impl#1}::drop (self=0x7fffe8000c50) at src/native_lib/event_queue.rs:139
#2  0x0000555555c7a877 in core::ptr::drop_in_place<wayland_client::imp::event_queue::EventQueueInner> () at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ptr/mod.rs:497
#3  0x0000555555c7b503 in alloc::rc::{impl#20}::drop<wayland_client::imp::event_queue::EventQueueInner> (self=0x7ffff7536f38) at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/rc.rs:1607
#4  0x0000555555c7ad4b in core::ptr::drop_in_place<alloc::rc::Rc<wayland_client::imp::event_queue::EventQueueInner>> () at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ptr/mod.rs:497
#5  0x0000555555bc1a97 in core::ptr::drop_in_place<wayland_client::event_queue::EventQueue> () at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ptr/mod.rs:497
#6  0x0000555555bcccc0 in smithay_clipboard::worker::worker_impl (display=..., request_rx=..., reply_tx=...) at src/worker/mod.rs:321
#7  0x0000555555bcb6dc in smithay_clipboard::worker::spawn::{closure#0} () at src/worker/mod.rs:44
…

### PR DESCRIPTION
**Pull request by kchibisov**
The main highlight is the move to calloop to handle all the requests which makes it possible to remove all the active polling and clean the data reading/writing, as well as accepting arbitrary long payloads.

This update also fixes the CI for the repository by moving it to github actions, sets the minimum rust version, and edition to 2021.

Fixes #46.
Fixes #44.
Fixes #34.

--

Depends on https://github.com/Smithay/client-toolkit/pull/416 and sctk release

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+            seat_state,
+            reply_tx,
+            seats,
+        })
+    }
+
+    /// Store selection for the given target.
+    ///
+    /// Selection source is only created when `Some(())` is returned.
+    pub fn store_selection(&mut self, ty: SelectionTarget, contents: String) -> Option<()> {
```
Comment (wash2) [inline src/state.rs:109]: Should this return a Result similar to `load_selection`?

File: src/worker.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+use sctk::reexports::calloop::channel::Channel;
+use sctk::reexports::calloop::{channel, EventLoop};
+use sctk::reexports::calloop_wayland_source::WaylandSource;
+use sctk::reexports::client::globals::registry_queue_init;
+use sctk::reexports::client::Connection;
+
+use crate::state::{SelectionTarget, State};
+
+/// Spawn a clipboard worker, which dispatches it's own `EventQueue` each 50ms
```
Comment (wash2) [inline src/worker.rs]: Does it dispatch every 50ms? Dispatch is called with `None`

Comment (kchibisov) [reply src/worker.rs]: It doesn't.

File: src/state.rs (hunk 1 above)
Comment (kchibisov) [reply src/state.rs:109]: The reason for that is that `load` wants a reply from the clipboard, so if we don't propagate the result and reply with `error`, we could hang the client, however the `store` doesn't need anything like that.

The only use case for store is to `log` something, which wasn't the case before.

I could document why one has and other doesn't return a result.

The reason I use `Option` is to simply rely on `?` to make the code less verbose.

Comment (wash2) [reply src/state.rs:109]: Ok, I figured it just makes it less verbose. I think it's fine as is, but just wanted to check

File: src/worker.rs (hunk 2 above)
Comment (wash2) [reply src/worker.rs]: Ok, I just ask because the comment says that it does each 50ms. I guess maybe it was copied from before the update.

Comment (kchibisov) [reply src/worker.rs]: Fixed. Also  pushed a fix for empty selection which I missed.

**Comment by kchibisov** [comment]
This should be good to go.

**Review by wash2 — APPROVED** [review]

**Pull request merged**